### ISSUE TITLE
Templates for inline code don't remove leading and trailing whitespace

### ISSUE THREAD

**Issue by Traumflug** [gohugoio/hugo#7309]
### What version of Hugo are you using?
0.71.0, the currently latest release.

----
This is a followup to https://github.com/gohugoio/hugo/issues/6832, which didn't receive a solution.

#### Regarding newlines at EOF

Having a newline at end of file isn't exactly [strange](https://github.com/gohugoio/hugo/issues/6832#issuecomment-587901259), but Unix standard. On Unix, text without a newline isn't a line, a text file without a newline inside is considered empty by many tools.

- Github agrees on this, it renders this red error mark at EOF without newline (e.g. [here](https://github.com/bep/hugo-issue-6832/commit/28a564c1a85d37aaee47b6235160ca51576c0547)).
- Command line Git agrees on this, it complains about the missing newlines at EOF.
- CLI tools like `wc` don't count text without newline.
- ...

That's why code editors usually add newlines at EOF automatically.

#### Regarding whitespace at BOF

Similar for leading whitespace: many code projects add copyright headers and similar stuff early in every file (e.g. [Hugo](https://github.com/gohugoio/hugo/blob/master/resources/image.go)). Hugo nicely supports this and removes comments for rendering, but this usually leaves some leading whitespace behind.

#### In short ...

Hugo should really remove whitespace at start and end of templates used inline. Just as if the whole file was wrapped in `{{-` and `-}}`.

**Comment by nekr0z** [comment]
Still an issue to consider, I suppose.

**Comment by bep** [comment]
>Still an issue to consider, I suppose.

Maybe, but you need to take this upstream in the Go project.

**Comment by nekr0z** [comment]
> Maybe, but you need to take this upstream in the Go project.

Ah, I see. Then we should indeed not attempt to solve it as a part of Hugo. Is there a way to label it as an upstream issue, just for the sake of documentation and clarity?

**Comment by bep** [comment]
But also note that I have seen many "template whitespace discussions" on that issue tracker, and I would not be too optimistic. T

**Comment by nekr0z** [comment]
I'd keep this one open, but I don't insist.
